docs/CreateAKSClusterForScalarDLAuditor.md

```diff
@@ -0,0 +1,215 @@
+
+### Create a node pool for monitoring components (kube-prometheus-stack and loki-stack)
+
+It is recommended to run only pods related to ScalarDL Ledger and ScalarDL Auditor on the worker node for ScalarDL Ledger and ScalarDL Auditor. If you want to run monitoring pods (Prometheus, Grafana, Loki, etc) by using [kube-prometheus-stack](./K8sMonitorGuide.md) and [loki-stack](./K8sLogCollectionGuide.md) on the same AKS cluster, you should create other node groups for monitoring pods.
+
+### Configure Cluster Autoscaler of AKS
+
+If you want to scale ScalarDL Ledger and ScalarDL Auditor pods automatically using [HPA (Horizontal Pod Autoscaler)](https://learn.microsoft.com/en-us/azure/aks/concepts-scale#horizontal-pod-autoscaler), you should configure Cluster Autoscaler of AKS too. Please refer to the [official document](https://learn.microsoft.com/en-us/azure/aks/concepts-scale#cluster-autoscaler) for more details.
+
+Also, if you configure Cluster Autoscaler, you should create a subnet in VNet for AKS to ensure a sufficient number of IPs to make AKS work without network issues after scaling. The required number of IPs is different depending on the networking plug-in. Please refer to the [kubenet document](https://learn.microsoft.com/en-us/azure/aks/configure-kubenet) and the [Azure CNI document](https://learn.microsoft.com/en-us/azure/aks/configure-azure-cni) for more details of the number of IPs.
+
+### Create the AKS cluster on a private network
+
+You should create the AKS cluster on a private network (private subnet in VNet) since ScalarDL Ledger and ScalarDL Auditor don't provide any services to users directly via internet access. It is better to access ScalarDL Ledger and ScalarDL Auditor via a private network from your applications.
+
+### Create the AKS cluster with Azure CNI if you need
+
+The default networking plug-in of AKS is [kubenet](https://learn.microsoft.com/en-us/azure/aks/configure-kubenet). If your requirement doesn't match kubenet, you should use the [Azure CNI](https://learn.microsoft.com/en-us/azure/aks/configure-azure-cni).
+
+For example, if you want to deploy multiple environments of ScalarDL Ledger and ScalarDL Auditor on one AKS cluster (e.g., deploy multi-tenant ScalarDL) and you want to control connection between each tenant using [NetworkPolicy](https://kubernetes.io/docs/concepts/services-networking/network-policies/), kubenet supports Calico only and [Calico is not supported by Azure support team](https://learn.microsoft.com/en-us/azure/aks/use-network-policies#differences-between-azure-npm-and-calico-network-policy-and-their-capabilities) (it is supported by Calico community or additional paid support).
```

Comment:
@josh-wong  
Ah, sorry. This is a very confusing thing...

From the technical perspective, we can deploy ScalarDL Ledger and Auditor on one AKS cluster as follows (this is one option for deployment of ScalaDL, but basically we cannot recommend it strongly. We should use this environment if and only if there are some special reason to use this environment).

```console
+--[One AKS cluster]----------------------------------------------------------+
|                                                                             |
|                                                                             |
|  +--[Namespace for Ledger]-----------------------------------------------+  |
|  |                                                                       |  |
|  |  +--[Worker Node 1]--+  +--[Worker Node 2]--+  +--[Worker Node 3]--+  |  |
|  |  |                   |  |                   |  |                   |  |  |
|  |  |  +-------------+  |  |  +-------------+  |  |  +-------------+  |  |  |
|  |  |  | Ledger Pod  |  |  |  | Ledger Pod  |  |  |  | Ledger Pod  |  |  |  |
|  |  |  +-------------+  |  |  +-------------+  |  |  +-------------+  |  |  |
|  |  |                   |  |                   |  |                   |  |  |
|  |  |  +-------------+  |  |  +-------------+  |  |  +-------------+  |  |  |
|  |  |  | Envoy Pod   |  |  |  | Envoy Pod   |  |  |  | Envoy Pod   |  |  |  |
|  |  |  +-------------+  |  |  +-------------+  |  |  +-------------+  |  |  |
|  |  |                   |  |                   |  |                   |  |  |
|  |  +-------------------+  +-------------------+  +-------------------+  |  |
|  |                                                                       |  |
|  +-----------------------------------------------------------------------+  |
|                                                                             |
|                                                                             |
|  +--[Namespace for Auditor]----------------------------------------------+  |
|  |                                                                       |  |
|  |  +--[Worker Node 4]--+  +--[Worker Node 5]--+  +--[Worker Node 6]--+  |  |
|  |  |                   |  |                   |  |                   |  |  |
|  |  |  +-------------+  |  |  +-------------+  |  |  +-------------+  |  |  |
|  |  |  | Auditor Pod |  |  |  | Auditor Pod |  |  |  | Auditor Pod |  |  |  |
|  |  |  +-------------+  |  |  +-------------+  |  |  +-------------+  |  |  |
|  |  |                   |  |                   |  |                   |  |  |
|  |  |  +-------------+  |  |  +-------------+  |  |  +-------------+  |  |  |
|  |  |  | Envoy Pod   |  |  |  | Envoy Pod   |  |  |  | Envoy Pod   |  |  |  |
|  |  |  +-------------+  |  |  +-------------+  |  |  +-------------+  |  |  |
|  |  |                   |  |                   |  |                   |  |  |
|  |  +-------------------+  +-------------------+  +-------------------+  |  |
|  |                                                                       |  |
|  +-----------------------------------------------------------------------+  |
|                                                                             |
|                                                                             |
+-----------------------------------------------------------------------------+
```

In this case, we separate Ledger and Auditor by `Namespace`, which is a Kubernetes feature instead of separating by AKS cluster.

We call this environment `multi-tenant ScalarDL`.

Also, for your information, I will create and add figures like the above in `Deploy ScalarDL Auditor on XXX` guide in another PR in the future.

Reply:
Looks good to me! I can see the important point `one AKS instead of two AKS` clearly.
Thank you!